front/app/components/CustomFieldsForm/SurveyFrom/index.tsx

```diff
@@ -0,0 +1,131 @@
+import React, { useRef, useState } from 'react';
+
+import { Box, Spinner } from '@citizenlab/cl2-component-library';
+
+import useCustomFields from 'api/custom_fields/useCustomFields';
+import useAddIdea from 'api/ideas/useAddIdea';
+import useDraftIdeaByPhaseId from 'api/ideas/useDraftIdeaByPhaseId';
+import useUpdateIdea from 'api/ideas/useUpdateIdea';
+import useAuthUser from 'api/me/useAuthUser';
+import { ParticipationMethod } from 'api/phases/types';
+import usePhase from 'api/phases/usePhase';
+import useProjectById from 'api/projects/useProjectById';
+
+import { updateSearchParams } from 'utils/cl-router/updateSearchParams';
+import { canModerateProject } from 'utils/permissions/rules/projectPermissions';
+
+import { convertCustomFieldsToNestedPages } from '../util';
+
+import SurveyPage, { FormValues } from './SurveyPage';
+
+const SurveyForm = ({
+  projectId,
+  phaseId,
+  participationMethod,
+}: {
+  projectId: string;
+  phaseId?: string;
+  participationMethod?: ParticipationMethod;
+}) => {
+  const pagesRef = useRef<HTMLDivElement | null>(null);
+  const [currentPageNumber, setCurrentPageNumber] = useState(0);
+
+  const { data: authUser } = useAuthUser();
+  const { data: project } = useProjectById(projectId);
+  const { data: phase } = usePhase(phaseId);
+  const { data: draftIdea, isLoading } = useDraftIdeaByPhaseId(phaseId);
+
+  const { mutateAsync: addIdea } = useAddIdea();
+  const { mutateAsync: updateIdea } = useUpdateIdea();
+  const { data: customFields } = useCustomFields({
+    projectId,
+    phaseId: participationMethod !== 'ideation' ? phaseId : undefined,
+  });
+
+  const nestedPagesData = convertCustomFieldsToNestedPages(customFields || []);
+
+  const showTogglePostAnonymously =
+    phase?.data.attributes.allow_anonymous_participation &&
+    participationMethod !== 'native_survey';
```

Comment:
What about this check? Also not sure what the plan is for the community monitor survey, whether we should show the toggle there, etc.

Reply:
This was not at all relevant for surveys so I deleted it!